docs/topics/gradle.md

```diff
@@ -722,6 +754,61 @@ See the [Gradle documentation](https://docs.gradle.org/current/userguide/configu
+> in [YouTrack](https://youtrack.jetbrains.com/issues/KT).
+>
+{type="warning"}
+
+Starting with [Kotlin 1.7.0](whatsnew17.md#build-reports-for-kotlin-compiler-tasks), build reports for tracking compiler
+performance are available. Reports contain the durations of different compilation phases and reasons why compilation
+couldn't be incremental.
+
+Use build reports to investigate performance issues, when the compilation time is too long or when it differs for the same
+project.
+
+To enable build reports, declare where to save the build report output in `gradle.properties`:
+
+```properties
+kotlin.build.report.output=file
+```
+
+The following values (and their combinations) are available:
+
+| Option       | Description                                                                                                                                                                                                                                                                                                                                     |
+|--------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `file`       | Saves build reports in a local file                                                                                                                                                                                                                                                                                                             |
+| `build_scan` | Saves build reports in the `custom values` section of the [build scan](https://scans.gradle.com/). Note that the Gradle Enterprise plugin limits the number of custom values and their length. In big projects, some values could be lost                                                                                                       |                                                                                                                                                                                                                                                                                                                                                                                               |
+| `http`       | Posts build reports using HTTP(S). The POST method sends metrics in the JSON format. You can see the current version of the sent data in the [Kotlin repository](https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/plugin/statistics/CompileStatisticsData.kt) |
+
+When analyzing build reports, you may encounter two cases:
```

Comment:
Not sure that explanation how to use is required here. We should write how to enable diagnostic tool, but not to teach people how to use it.